src/uct/tcp/tcp_cm.c

```diff
@@ -663,6 +658,15 @@ void uct_tcp_cm_handle_conn_ack(uct_tcp_ep_t *ep, uct_tcp_cm_conn_event_t cm_eve
     }
 }
 
+static void uct_tcp_cm_handle_conn_fin(uct_tcp_ep_t *ep)
+{
+    if ((ep->flags & UCT_TCP_EP_CTX_CAPS) == UCT_TCP_EP_FLAG_CTX_TYPE_RX) {
+        uct_tcp_ep_set_failed(ep);
```

Comment:
why not just call uct_tcp_ep_destroy_internal()?
since the ep is already destroyed from user perspective (no TX), IMO no need to go through set_ep_failed

Reply:
it was done to not dereference invalid EP from the progress function.
done - in the same way as for `REQ`.